src/pydata_sphinx_theme/assets/styles/variables/_color.scss

```diff
@@ -1,21 +1,46 @@
+/* for colored buttons and badges, ensure text color is legible */
+@function text-color($value) {
+  @return if(lightness($value) > 70, #000, #fff);
+}
+
 /*******************************************************************************
 * light theme
 *
 * all the variables used for light theme coloring
 */
 html[data-theme="light"] {
   /*****************************************************************************
-  * main colors
+  * main colors. the $pst-semantic-colors follow the naming convention in
+  * sphinx-design, to allow for color overrides in
+  * ../extensions/_sphinx_design.scss. NOTE: some of these colors (most notably
+  * the *-text and *-highlight variables) are **only** included to allow
+  * sphinx-design overrides, and (probably?) should not be used in our own CSS rules.
+  *
+  * NOTE: this theme defines "info == primary" and "warning == secondary"
   */
-  --pst-color-primary: rgb(69 157 185);
-  --pst-color-secondary: rgb(238 144 64);
-  --pst-color-success: rgb(40, 167, 69);
-  --pst-color-info: var(--pst-color-primary);
-  --pst-color-warning: var(--pst-color-secondary);
-  --pst-color-danger: rgb(220, 53, 69);
-  --pst-color-text-base: rgb(51, 51, 51);
-  --pst-color-text-muted: rgb(77, 77, 77);
-  --pst-color-border: rgb(201, 201, 201);
+  $pst-semantic-colors: (
+    "primary": rgb(69, 157, 185),
+    "secondary": rgb(238, 144, 64),
+    "success": rgb(40, 167, 69),
+    "info": rgb(69, 157, 185),
+    "warning": rgb(238, 144, 64),
+    "danger": rgb(220, 53, 69),
+    "light": rgb(201, 201, 201),
+    "muted": rgb(77, 77, 77),
+    "dark": rgb(51, 51, 51),
+    "black": rgb(0, 0, 0),
+    "white": rgb(255, 255, 255),
+  );
+
+  @each $name, $value in $pst-semantic-colors {
+    --pst-color-#{$name}: #{$value};
+    --pst-color-#{$name}-text: #{text-color($value)};
+    --pst-color-#{$name}-highlight: #{mix(#000, $value, 15%)};
+  }
+  /* ↓↓↓ these are not sphinx-design overrides ↓↓↓ */
```

Comment:
Hmm, I am worried about intermixing "special casing" CSS with "default CSS" in one place, I feel like this could be confusing given that Sphinx Design is not bundled with the theme.

Is there a way that we can cleanly separate the two so that one section is entirely "just this theme" and another section is clearly "sphinx design", or even better if we can put the sphinx design stuff *just* in the `extensions/` folder?